### PR TITLE
RUMM-221 Webview Integrations

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Provide RUM/Logs Webview Integrations

### Review checklist (to be filled by reviewers)

- [x] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [x] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [x] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt [hunk 1]
```diff
@@ -0,0 +1,121 @@
...
+        errorCode: Int,
+        description: String?,
+        failingUrl: String?
+    ) {
+        super.onReceivedError(view, errorCode, description, failingUrl)
+        GlobalRum.get().addError(
+            "Error $errorCode: $description",
+            ORIGIN,
+            null,
+            mapOf(RumEventSerializer.TAG_HTTP_URL to failingUrl)
```
Comment (mariusc83) [inline dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt]: Hmm not a big fan of adding this key from the RumEventSerializer. I would rather add a new constant here in this class and remove don't mix the RumEventSerializer here. It feels like this doesn't belong here.

File: dd-sdk-android/src/test/kotlin/com/datadog/android/rum/RumWebViewClientTest.kt [hunk 2]
```diff
@@ -0,0 +1,334 @@
...
+            "Error $errorCode: $description",
+            RumWebViewClient.ORIGIN,
+            null,
+            mapOf(RumEventSerializer.TAG_HTTP_URL to null)
+        )
+    }
+
+    @Test
+    @TestTargetApi(Build.VERSION_CODES.M)
+    fun `onReceivedError2 sends a Rum Error`(
```
Comment (mariusc83) [inline dd-sdk-android/src/test/kotlin/com/datadog/android/rum/RumWebViewClientTest.kt]: maybe a more suggestive name here ? "onReceivedError on SDK 23..."

File: sample/kotlin-timber/src/main/kotlin/com/datadog/android/sample/webview/WebFragment.kt [hunk 3]
```diff
@@ -40,8 +29,8 @@ class WebFragment : Fragment() {
     ): View? {
         val rootView = inflater.inflate(R.layout.fragment_web, container, false)
         webView = rootView.findViewById(R.id.webview)
-        webView.webViewClient = mWebViewClient
-        webView.webChromeClient = webChromeClient
+        webView.webViewClient = RumWebViewClient()
```
Comment (mariusc83) [inline sample/kotlin-timber/src/main/kotlin/com/datadog/android/sample/webview/WebFragment.kt]: Nice ;)

File: dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt (hunk 1 above)
Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt]: The thing is that this is a standard key. Duplicating the constant makes it more error prone (if we change the field in RumEventSerializer we might forget to change it here or vice versa). 
The goal of a constant is to be constant everywhere.

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt]: Still...this tells me that if I want to use the RumMonitor API (by passing some attributes) I need to know the internal classes (e.g. RumEventSerializer) and its internal tags. Also what will happen if for example you want to extract this in another module (artifact). You will not be able to do that without exposing the RumEventSerializer class as this is an internal. Maybe instead we should add new method inside the RumMonitor -> `addError(message, origin, failingUrl, attributes)`

Comment (xgouchet) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt]: This is our own implementation, so we do know the internals. People using the RUM monitor can use whatever attributes they want, they won't look into this.
If I need to extract it into a different module, then I'll duplicate the constant.

No let's not add this new method. Errors don't necessarily have to do with urls, so this is a specific use case.

Comment (mariusc83) [reply dd-sdk-android/src/main/kotlin/com/datadog/android/rum/RumWebViewClient.kt]: Nevermind...I will accept it but still not very convinced.

**Pull request merged**